SimPEG/Directives.py

```diff
@@ -588,6 +597,28 @@ def endIter(self):
                 self.invProb.beta = (self.invProb.beta * self.target /
                                      self.invProb.phi_d)
 
+    def validate(self, directiveList):
+        # check if a linear preconditioner is in the list, if not warn else
+        # assert that it is listed after the IRLS directive
+        dList = directiveList.dList
+        self_ind = dList.index(self)
+        lin_precond_ind = [
+            isinstance(d, Update_lin_PreCond) for d in dList
+        ]
+
+        if any(lin_precond_ind):
```

Comment:
@fourndo: can you take a look at this? Here I check if the there is a `Update_lin_PreCond` directive in the directive list and then make sure it is after the IRLS directive or throw a warning if not in the list (not an error - see what you think is more appropriate)